pkg/nad/masterplugin.go

```diff
@@ -393,3 +393,161 @@ func (plugin *MasterIPVlanPlugin) GetMasterPluginConfig() (*MasterPlugin, error)
 
 	return plugin.masterPlugin, nil
 }
+
+// MasterBondPlugin provides struct for MasterPlugin to create a NAD for bonded interfaces.
+type MasterBondPlugin struct {
+	masterPlugin *MasterPlugin
+	errorMsg     string
+}
+
+// NewMasterBondPlugin define network-attachment-interface for bond interface.
+func NewMasterBondPlugin(name string) *MasterBondPlugin {
+	glog.V(100).Infof(
+		"Initializing new MasterBondPlugin structure %s", name)
+
+	builder := MasterBondPlugin{
+		masterPlugin: &MasterPlugin{
+			CniVersion: "0.3.1",
+			Name:       name,
+			Type:       "bond",
+		},
+	}
+
+	if builder.masterPlugin.Name == "" {
+		glog.V(100).Infof("error NewMasterBondPlugin name can not be empty")
+
+		builder.errorMsg = "NewMasterBondPlugin name is empty"
+	}
+
+	return &builder
+}
+
+// WithMode defines and sets the bond mode type.
+func (plugin *MasterBondPlugin) WithMode(mode string) *MasterBondPlugin {
+	validModes := map[string]bool{
+		"balance-rr":    true,
+		"active-backup": true,
+		"balance-xor":   true,
+	}
+
+	if mode == "" {
+		glog.V(100).Infof("error adding empty mode to MasterBondPlugin")
+
+		return nil
```

Comment:
no need return here, add error to the struct
plugin.errorMsg ="Bond mode can not be empty"

Reply:
changed